src/compiler/Parser/StringParser/StringParser.ts

```diff
@@ -303,8 +303,12 @@ export class StringParser {
     return null;
   };
 
-  public readonly Optional = (rule: ParseRule): ParseRule => () =>
-    this.ParseObject(rule) || StringParser.ParseSuccess;
+  public readonly Optional = (rule: ParseRule): ParseRule => () => {
```

Comment:
```
'' || ParseSuccess
```
returned `ParseSuccess`, which prevented the CommentEliminator to remove `/* inline block comments*/`. This is the only thing in the code that correctly translate to an empty string.